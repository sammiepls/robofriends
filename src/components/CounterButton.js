import React, {useState} from "react";

class CounterButton extends React.PureComponent {
  constructor() {
    super();
    this.state = {
      count: 0,
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.state.count !== nextState.count ? true : false
  }

  updateCount = () => {
    this.setState(state => ({ count: state.count + 1 }));
  };

  render() {
    return (
      <button color={this.props.color} onClick={this.updateCount}>
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;

// const CounterButton = props => {
//   const [count, setCount] = useState(0)

//   return (
//       <button color={props.color} onClick={() => setCount(prevState => prevState + 1)}>
//         Count: {count}
//       </button>
//     );
// }

// export default CounterButton
