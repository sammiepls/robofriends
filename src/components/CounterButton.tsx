// @ts-nocheck
import React from "react";

class CounterButton extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }

  shouldComponentUpdate(nextProps, nextState) {
    return this.state.count !== nextState.count ? true : false
  }

  updateCount = () => {
    this.setState(state => ({ count: state.count + 1 }));
  };

  render() {
    return (
      <button color={this.props.color} onClick={this.updateCount}>
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;

// import React, { useState } from "react";

// const CounterButton = () => {
//   const [count, setCount] = useState(0)

//   return (
//       <button onClick={() => setCount(prevState => prevState + 1)}>
//         Count: {count}
//       </button>
//     );
// }

// export default CounterButton
