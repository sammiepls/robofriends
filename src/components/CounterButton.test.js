import React from 'react';
import {shallow} from 'enzyme'
import CounterButton from './CounterButton'

describe('test CounterButton', () => {
  let wrapper;

  beforeEach(() => {
     wrapper = shallow(<CounterButton/>)
  })

  it('should render the CounterButton component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // it('should increate the count when button is pressed', () => {
  //   expect(wrapper.text()).toBe('Count: 0')
  //   const button = wrapper.find('button');
  //   button.simulate('click');
  //   expect(wrapper.text()).toBe('Count: 1')
  // })

  // class component
  it("should increase the count when button is pressed", () => {
    expect(wrapper.state(["count"])).toEqual(0);
    wrapper.simulate("click");
    expect(wrapper.state(["count"])).toEqual(1);
  });
})
