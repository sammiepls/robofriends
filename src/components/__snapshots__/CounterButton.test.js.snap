// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CounterButton should render the CounterButton component 1`] = `
<button
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
